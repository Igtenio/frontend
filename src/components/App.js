import React, { Component } from 'react';

import { connect } from 'react-redux';

import { Container, Row } from 'react-bootstrap';

import Header from './templates/Header';

import Footer from './templates/Footer';

import { Router } from '@reach/router';

import Calculator from './pages/Calculator';

import Flavors from './pages/Flavors';

import Home from './pages/Home';

import Profile from './pages/user/Profile';
import UserRecipes from './pages/user/Recipes';
import Favorites from './pages/user/Favorites';
import FlavorStash from './pages/user/FlavorStash';
import UserSettings from './pages/user/Settings';

import Login from './pages/Login';

import Register from './pages/Register';

export class App extends Component {
  render() {
    return (
      <Container>
        <Header />
        <Row>
          <Router primary={false}>
            <Home path="/" />
            <Profile path="profile" />
            <UserRecipes path="userRecipes" />
            <Favorites path="favorites" />
            <FlavorStash path="flavorStash" />
            <UserSettings path="userSettings" />
            <Login path="login" />
            <Register path="register" />
            <Calculator path="calculator" />
            <Flavors path="flavors" />
          </Router>
        </Row>
        <Footer />
      </Container>
    );
  }
}

export default connect(
  null,
  null
)(App);
