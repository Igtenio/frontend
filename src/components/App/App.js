import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import React, { Component, Fragment } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';

import Header from 'components/Header/Header';

import Home from 'pages/Home';
import Login from 'pages/Login';
import NotFound from 'pages/NotFound';
import Register from 'pages/Register';

import { actions as appActions } from 'reducers/application';

export class App extends Component {
  static propTypes = {
    actions: PropTypes.object.isRequired
  };

  componentDidMount() {
    const { actions } = this.props;

    actions.initApp();
  }

  render() {
    return (
      <Fragment>
        <Header />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/register" component={Register} />
          <Route component={NotFound} />
        </Switch>
      </Fragment>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      ...appActions
    },
    dispatch
  )
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(App)
);
