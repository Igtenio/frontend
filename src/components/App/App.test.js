import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import App from './App';
import { withMemoryRouter } from 'utils';

describe('<App />', () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const RoutedApp = withMemoryRouter(App);
    const component = renderer.create(
      <Provider store={store}>
        <RoutedApp />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
