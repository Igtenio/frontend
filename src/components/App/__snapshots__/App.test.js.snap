// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <span
      className="navbar-brand"
    >
      MixNJuice
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <h1>
      Home
    </h1>
  </div>,
]
`;
